### PR TITLE
update packages

### PR DESCRIPTION
**Pull request by icewind1991**
Update the packages and adjust to the now-required java21

### CODE REVIEW

File: piped-backend/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ src, stdenv, fetchFromGitHub, lib, gradle, perl, runtimeShell, jdk19_headless, nixosTest }:
+{ src, stdenv, fetchFromGitHub, lib, gradle, perl, runtimeShell, jdk, nixosTest, dos2unix }:
```
Comment (squalus) [inline piped-backend/default.nix:1]: Renaming the input to `jdk` is a good change.

File: piped-backend/default.nix [hunk 2]
```diff
@@ -71,8 +71,11 @@ HERE
 
     inherit src;
 
-    nativeBuildInputs = [ gradle ];
+    nativeBuildInputs = [ gradle dos2unix ];
 
+    prePatch = ''
+      dos2unix src/main/java/me/kavin/piped/Main.java
```
Comment (squalus) [inline piped-backend/default.nix:77]: The patch should be a diff on the unmodified upstream file. (Unless this is a pre-existing nixpkgs convention that I'm unware of).

**Review by squalus — CHANGES_REQUESTED** [review]
Thanks for the PR. This mostly matches the unfinished work I have. `piped-test` fails now, though.

```
vm-test-run-piped-integration-test> machine # [  925.063666] postgres[8869]: [8869] ERROR:  permission denied for schema public at character 14
vm-test-run-piped-integration-test> machine # [  925.064087] postgres[8869]: [8869] STATEMENT:  CREATE TABLE public.databasechangelog (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP WITHOUT TIME ZONE NOT NULL, ORDEREXECUTED INTEGER NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10))

```

File: piped-backend/default.nix (hunk 2 above)
Comment (icewind1991) [reply piped-backend/default.nix:77]: The source seems to be in dos line endings, and having the patch be in dos line endings wasn't working for me, `patch` would just print an "ignoring extra \r at end of line" and than fail to apply it.

Converting the line endings of the patched file was the only way I got it to apply.

**Comment by icewind1991** [comment]
Fixed the postgresql issue with the tests, `piped-test` still fails for me though since it requires internet access

**Comment by squalus** [comment]
I no longer see the permissions error in piped-test.

piped-test has to be run with the sandbox turned off because of the internet access issue, like this: `nix build -L .#piped-test --option build-use-sandbox false`.

piped-test still fails for me. When trying it out in the interactive machine, like with `nix build -L .#piped-test.driverInteractive`, it's a blank page in chromium. The network requests to piped-backend hang.

This hangs: `curl -v 'http://127.0.0.1:14302/trending?region=US'`

Did you perform any manual testing on this, and did it work fine for you?

**Comment by icewind1991** [comment]
> Did you perform any manual testing on this, and did it work fine for you?

I've been using a [version of this that uses mostly stable nixpkgs](https://github.com/icewind1991/piped-flake/tree/stable23.05) for over a week without issues.

**Comment by squalus** [comment]
I fixed the piped-test hang by increasing cores from 1 to 3. Piped-Backend was hanging in the DownloaderImpl while NewPipeExtractor was trying to fetch the sw.js file, on the newVirtualThreadPerTaskExecutor. Seems like some bug on the Piped-Backend side.

**Pull request merged**